src/client/call-composer.ts

```diff
@@ -0,0 +1,116 @@
+import { DecIndent, DecIndentAndCloseBlock, DocumentParts, IncIndent } from '../output/writer'
+import { GeneratorContext } from './generator-context'
+import * as algokit from '@algorandfoundation/algokit-utils'
+import { makeSafeMethodIdentifier } from '../util/sanitization'
+import { BARE_CALL, MethodList } from './helpers/get-call-config-summary'
+import { getCreateOnCompleteOptions } from './deploy-types'
+
+export function* composeMethod(ctx: GeneratorContext): DocumentParts {
+  const { name, callConfig } = ctx
+  yield `public compose(): ${name}Composer {`
+  yield IncIndent
+
+  yield `const client = this`
+  yield `const atc = new AtomicTransactionComposer()`
+  yield `let promiseChain:Promise<unknown> = Promise.resolve()`
+  yield `const resultMappers: Array<undefined | ((x: any) => any)> = []`
+  yield `return {`
+  yield IncIndent
+
+  yield* callComposerNoops(ctx)
+  yield* callComposerOperationMethods(ctx, callConfig.updateMethods, 'update', true)
```

Comment:
@robdmoore I have omitted 'create' from this list as to my understanding, you cannot create and interact with a smart contract in one transaction as you can't know the application id until it is created

Reply:
Create might still be valid in the context of other app instances, for example when deploying we create a new app, and delete the old app in the same transaction.

Reply:
Sorry, I wasn't very clear, `deploy` was just an example of creating an app and doing something else in a transaction

In the python typed client, it's possible to do something like this

```
atc = AtomicTransactionComposer()
(app_a_client.compose(atc) 
        .hello(name="world")
        .hello_check(name="bar")
)
(app_b_client.compose(atc)
        .foo(bar="baz")
        .interesting(use="case")
)
app_c_client.compose(atc).create()
atc.execute()
```
And it will do attempt to do all 5 transactions in the one group

So while it isn't possible to interact with a newly created app in the ATC, it could be possible to create one and do other things

Reply:
Ah yeah that makes sense. That usage model would probably break the fancy type inference from Tristan (as might an addTransaction) escape hatch come to think of it?